catroid/src/main/java/org/catrobat/catroid/ui/ProjectUploadActivity.java

```diff
@@ -263,6 +278,62 @@ private void onNextButtonClick() {
 		}
 	}
 
+	private void checkCodeforApiKey() {
+		String regex = "(<value>).*?((?=.*[A-Za-z].*[0-9].*)[A-Za-z0-9]{24,45}).*?(</value>)";
```

Comment:
```suggestion
		String regex = "<value>.*?((?=[A-Za-z]+[0-9]|[0-9]+[A-Za-z])[A-Za-z0-9]{24,45})";
```

Note: the capturing group needs to be changed to 1 in line 294 below.